src/notebooks/controllers/notebookControllerManager.ts

```diff
@@ -673,7 +673,7 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
                     traceVerbose(`Early registration of controller for Kernel connection ${preferredConnection.id}`);
                     this.createNotebookControllers([preferredConnection]);
                 }
-            } else {
+            } else if (document.notebookType === InteractiveWindowView) {
```

Comment:
This was the bug fix, its not a flaky test issue, even when we had a preferred controller we'd change it to the active interpreter.